ash-swapchain/examples/demo.rs

```diff
@@ -0,0 +1,310 @@
+                    .new_layout(vk::ImageLayout::TRANSFER_DST_OPTIMAL)
+                    .image(swapchain_image)
+                    .subresource_range(vk::ImageSubresourceRange {
+                        aspect_mask: vk::ImageAspectFlags::COLOR,
+                        base_mip_level: 0,
+                        level_count: 1,
+                        base_array_layer: 0,
+                        layer_count: 1,
+                    })
+                    .build()],
+            );
+            let t = (self.epoch.elapsed().as_secs_f32().sin() + 1.0) * 0.5;
+            self.device.cmd_clear_color_image(
+                cmd,
+                swapchain_image,
+                vk::ImageLayout::TRANSFER_DST_OPTIMAL,
+                &vk::ClearColorValue {
+                    float32: [0.0, t, 0.0, 1.0],
+                },
+                &[vk::ImageSubresourceRange {
+                    aspect_mask: vk::ImageAspectFlags::COLOR,
+                    base_mip_level: 0,
+                    level_count: 1,
+                    base_array_layer: 0,
+                    layer_count: 1,
+                }],
+            );
+            // Typically this barrier would be implemented with an implicit subpass dependency to
+            // EXTERNAL
+            self.device.cmd_pipeline_barrier(
+                cmd,
+                vk::PipelineStageFlags::TRANSFER,
+                vk::PipelineStageFlags::BOTTOM_OF_PIPE,
+                vk::DependencyFlags::default(),
+                &[],
+                &[],
+                &[vk::ImageMemoryBarrier::builder()
+                    .src_access_mask(vk::AccessFlags::TRANSFER_WRITE)
+                    .src_queue_family_index(vk::QUEUE_FAMILY_IGNORED)
+                    .dst_queue_family_index(vk::QUEUE_FAMILY_IGNORED)
+                    .old_layout(vk::ImageLayout::TRANSFER_DST_OPTIMAL)
+                    .new_layout(vk::ImageLayout::PRESENT_SRC_KHR)
+                    .image(swapchain_image)
+                    .subresource_range(vk::ImageSubresourceRange {
+                        aspect_mask: vk::ImageAspectFlags::COLOR,
+                        base_mip_level: 0,
+                        level_count: 1,
+                        base_array_layer: 0,
+                        layer_count: 1,
+                    })
+                    .build()],
+            );
+
+            //
+            // Submit commands and queue present
+            //
+
+            self.device.end_command_buffer(cmd).unwrap();
+            self.device
+                .queue_submit(
+                    self.queue,
+                    &[vk::SubmitInfo::builder()
+                        .wait_semaphores(&[acq.ready])
+                        .wait_dst_stage_mask(&[vk::PipelineStageFlags::TRANSFER])
+                        .signal_semaphores(&[self.frames[acq.frame_index].complete])
+                        .command_buffers(&[cmd])
+                        .build()],
```

Comment:
When needing a slice of a single element that is created by a builder, I typically use `std::slice::as_ref()` which allows the builder to `Deref` without calling `.build()`.

Reply:
That seems like extraneous ceremony to me, since this `build` call is lexically within the statement that consumes it and therefore sound regardless.